### DISCUSSION TITLE
Move code blocks to the right-hand side of the page

### DISCUSSION THREAD

**Discussion by dextertd** [facebook/docusaurus#8224] [Q&A]
For the purposes of documenting APIs, it makes sense to display large amounts of code to the right-hand side of the main body of the document, in order to fit in as much content as possible.

<img width="932" alt="Screenshot 2022-10-18 at 16 31 28" src="https://user-images.githubusercontent.com/54495183/196475970-d034e90a-a9d6-48d4-84e1-2f3b9120ea3f.png">

How can this be achieved with Docusaurus? I have found several plugins which do this ([Redocusaurus](https://github.com/rohit-gohri/redocusaurus), [docusaurus-openapi-docs](https://github.com/PaloAltoNetworks/docusaurus-openapi-docs)), but they are generating complex mdx files from swagger OpenAPI specs. I would like to apply this to my own plain mdx files (which are not generated, but written).

I cannot find a plugin for this but can anyone point me in the right direction? Thanks.

**Comment by slorber** [comment]
Courier has built something using plain hand-written MDX and React components. 

https://www.courier.com/blog/how-we-built-our-documentation/

https://www.courier.com/docs/reference/send/message/

<img width="1913" alt="CleanShot 2022-10-19 at 14 59 29@2x" src="https://user-images.githubusercontent.com/749374/196697625-92063911-5362-4a69-9090-ed2f268b8c75.png">


Their site is open-source and afaik they don't use a Docusaurus plugin to create that API client: https://github.com/trycourier/docs
